compiler/lib/generate.ml

```diff
@@ -674,12 +675,23 @@ let build_graph ctx pc =
   let visited_blocks = ref Addr.Set.empty in
   let loops = ref Addr.Set.empty in
   let succs = Hashtbl.create 17 in
+  let poptrap = Hashtbl.create 17 in
   let backs = Hashtbl.create 17 in
   let preds = Hashtbl.create 17 in
   let seen = Hashtbl.create 17 in
+  let cf_frontier = ref Addr.Map.empty in
   let blocks = ctx.Ctx.blocks in
   let dominance_frontier_cache = Hashtbl.create 17 in
-  let rec loop pc anc =
+  let incr_prec pc =
+    match Hashtbl.find preds pc with
+    | exception Not_found -> Hashtbl.add preds pc 1
+    | n -> Hashtbl.replace preds pc (succ n)
+  in
+  let add_cf_frontier pc front =
+    Addr.Set.iter incr_prec front;
+    cf_frontier := Addr.Map.add pc front !cf_frontier
```

Comment:
Could we just add and edge in `succs` from the block containing the `Pushtrap` to right after the `Poptrap` instead?

Reply:
https://github.com/ocsigen/js_of_ocaml/pull/1344/commits/5ba840f8f02f5a263c58d3095f9fb1e727da3d55 does that